File: TypeScript/Practice/LeetCode/Easy/Unique Number of Occurrences.ts
// https://leetcode.com/problems/unique-number-of-occurrences/description/?envType=daily-question&envId=2024-01-17

function uniqueOccurrences(arr: number[]): boolean {
    const numbersFrequency: {[key: string]: number} = {};

    for (let i = 0; i < arr.length; i++) {
        if (numbersFrequency[arr[i]] === undefined) {
            numbersFrequency[arr[i]] = 1;
        } else {
            numbersFrequency[arr[i]] += 1;
        }
    }

    return Object.values(numbersFrequency).length === new Set(Object.values(numbersFrequency)).size;    
};

File: C#/Practice/LeetCode/Easy/Unique Number of Occurrences.cs
// https://leetcode.com/problems/unique-number-of-occurrences/description/?envType=daily-question&envId=2024-01-17

public class Solution {
    public bool UniqueOccurrences(int[] arr) {
        Dictionary<int, int> numbersFrequency = new Dictionary<int, int>();

        foreach (int num in arr)
        {
            if (!numbersFrequency.ContainsKey(num))
            {
                numbersFrequency[num] = 1;
            }
            else
            {
                numbersFrequency[num]++;
            }
        }

        bool areAllValuesUnique = numbersFrequency.Values.Distinct().Count() == numbersFrequency.Values.Count();

        return areAllValuesUnique;
    }
}